src/Worker/Grpc/GrpcDurableTaskWorker.Processor.cs

```diff
@@ -237,6 +241,14 @@ async Task ProcessWorkItemsAsync(AsyncServerStreamingCall<P.WorkItem> stream, Ca
                         this.Logger.UnexpectedWorkItemType(workItem.RequestCase.ToString());
                     }
                 }
+
+                if (tokenSource.IsCancellationRequested || tokenSource.Token.IsCancellationRequested)
+                {
+                    // The token has cancelled, this means either:
+                    // 1. The broader 'cancellation' was triggered, return here to start a graceful shutdown.
+                    // 2. The timeoutSource was triggered, return here to trigger a reconnect to the backend.
```

Comment:
Can we add a warning log when we detect that we hit the timeout condition? I feel like that would be a good distinction for us to have some telemetry on.

Reply:
Added a log.